e2e/tests/golden-tests.rs

```diff
@@ -0,0 +1,80 @@
+use insta::assert_debug_snapshot;
+use postgres::SimpleQueryMessage;
+use std::env;
+use std::fs::read_to_string;
+use std::path::Path;
+use test_generator::test_resources;
+use testcontainers::images::generic::WaitFor;
+use testcontainers::{clients, images, Docker};
+
+const DB: &str = "postgres-db-test";
+const USER: &str = "postgres-user-test";
+const PASSWORD: &str = "postgres-password-test";
+
+#[test_resources("testdata/*.sql")]
+fn golden_test(resource: &str) {
+    let _ = pretty_env_logger::try_init();
+
+    let docker = clients::Cli::default();
+
+    let docker_image = env::var("TS_DOCKER_IMAGE").unwrap();
+
+    let generic_postgres = images::generic::GenericImage::new(docker_image)
+        .with_wait_for(WaitFor::message_on_stderr(
+            "database system is ready to accept connections",
+        ))
+        .with_env_var("POSTGRES_DB", DB)
+        .with_env_var("POSTGRES_USER", USER)
+        .with_env_var("POSTGRES_PASSWORD", PASSWORD);
+
+    let node = docker.run(generic_postgres);
+
+    let connection_string = &format!(
+        "postgres://{}:{}@localhost:{}/{}",
+        USER,
+        PASSWORD,
+        node.get_host_port(5432).unwrap(),
+        DB
+    );
+
+    let mut client = postgres::Client::connect(connection_string, postgres::NoTls).unwrap();
+
+    let path = Path::new(resource);
+    assert!(path.exists());
+    let query = read_to_string(path).unwrap();
+    let result = client.simple_query(&query).unwrap();
```

Comment:
It would be nicer to just run `docker exec psql -c ...` and take the output of that. At the moment testcontainers doesn't support that, so I've taken the (annoyingly hacky) approach of running the file as a query against the DB and then transforming the results to a string.

The formatting of the final string could be improved to increase readability.

Reply:
Why don't we just do `docker exec psql -c ...` using process::Command::new("docker")... that's what testcontainers does anyway then 
```
let res = process::Command::new("docker")
            .arg("exec")
            .arg("-d")
            .arg(id)
            .arg("sh")
            .arg("-c")
            .arg(cmd)
            .stdout(Stdio::piped())
             .output()
             .expect("could not exec");
String::from_utf8_lossy(&output.stdout)

```

Reply:
Thanks for the input, I've done it this way.